flair/datasets/sequence_labeling.py

```diff
@@ -592,6 +595,87 @@ def __offset_docstarts(file_in: Union[str, Path], file_out: Union[str, Path]):
                     f.write("\n")
 
 
+
+class ICELANDIC_NER(ColumnCorpus):
+    def __init__(
+            self,
+            base_path: Union[str, Path] = None,
+            tag_to_bioes: str = "ner",
+            in_memory: bool = True,
+            **corpusargs,
+    ):
+        """
+        Initialize the ICELANDIC_NER corpus. The first time you call this constructor it will automatically
+        download the dataset.
+        :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
+        to point to a different folder but typically this should not be necessary.
+        :param tag_to_bioes: NER by default, need not be changed, but you could also select 'pos' to predict
+        POS tags instead
+        :param in_memory: If True, keeps dataset in memory giving speedups in training.
+        :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
+        """
+        if type(base_path) == str:
+            base_path: Path = Path(base_path)
+
+        # column format
+        columns = {0: "text", 1: "ner"}
+
+        # this dataset name
+        dataset_name = self.__class__.__name__.lower()
+
+        # default dataset folder is the cache root
+        if not base_path:
+            base_path = Path(flair.cache_root) / "datasets"
+        data_folder = base_path / dataset_name
+
+        # download data if necessary
+        ZipFile.extractall(path=icelandic_ner, members="https://repository.clarin.is/repository/xmlui/handle/20.500.12537/42/allzip", pwd=None) 
+        entries = os.listdir('icelandic_ner')
+
+        with open("icelandic_ner_path/train.txt", "w") as outfile:
+            for filename in entries:
+                with open(filename) as infile:
+                    contents = infile.read()
+                    outfile.write(contents)
+
+        # download files if not present locally
+        cached_path(f"{icelandic_ner_path}ned.testa", data_folder / 'raw')
+        cached_path(f"{icelandic_ner_path}ned.testb", data_folder / 'raw')
+        cached_path(f"{icelandic_ner_path}ned.train", data_folder / 'raw')
+
+        # we need to slightly modify the original files by adding some new lines after document separators
+        train_data_file = data_folder / 'train.txt'
+        if not train_data_file.is_file():
```

Comment:
Why is this part necessary? Are extra offsets needed? Maybe you can use the files as they are?

Reply:
I have changed it this morning please can you check it again?